algorithms/integration/processor.py

```diff
@@ -459,44 +459,69 @@ def __call__(self):
             self.params.debug.output,
         )
 
-        # Compute expected memory usage and warn if not enough
-        total_memory = psutil.virtual_memory().total
-        sbox_memory = processor.compute_max_memory_usage()
         assert (
             self.params.block.max_memory_usage > 0.0
         ), "maximum memory usage must be > 0"
         assert (
             self.params.block.max_memory_usage <= 1.0
         ), "maximum memory usage must be <= 1"
-        limit_memory = total_memory * self.params.block.max_memory_usage
-        if sbox_memory > limit_memory:
+
+        # Compute expected memory usage and warn if not enough
+        required_memory = processor.compute_max_memory_usage()
+        available_memory = psutil.virtual_memory().available
```

Comment:
I note that VM on fruiters is a bit special - it is not a static amount of swap, more something where it grows according to it's own whims - 

```
 Memory usage:
  Available system memory (excluding swap): 9.2 GB
  Available swap memory: 1.6 GB
  Available system memory (including swap): 10.8 GB
  Limit shoebox memory (including swap): 1.2 GB
  Limit shoebox memory (excluding swap): 1.0 GB
  Required shoebox memory: 0.0 GB
```

is pretty misleading

Reply:
Yeah, dynamic swap allocation is a thing. Then again we don't really want to work in swap anyway. An alternative would be to give a warning, but then continue and potentially run into `-9` if the turing-tape isn't actually infinite.

Reply:
Our prior on Turing tapes is that they are finite, so the change set is probably good. Am wondering then whether we should consider the swap at all? I guess if one process requires more memory than we have currently free in wired + swap then we are in trouble, but beyond that does it contain much signal?

At the same time, the signal-to-noise of the log output is pretty low, and rarely read, so on balance does no harm

Reply:
I used swap vs ram mainly in deciding how many processes to allow, and the sum in deciding when to give up, so
10 GB RAM, 10 GB swap, 1 GB/process => maximum number of processes: 10
10 GB RAM, 10 GB swap, 10 GB/process => maximum number of processes: 1
10 GB RAM, 10 GB swap, 20 GB/process => maximum number of processes: 1
10 GB RAM, 10 GB swap, 20.1 GB/process => fail with error message